### PR TITLE
Update packages

### PR DESCRIPTION
**Pull request by wilwade**
## Goal

The goal of this PR is to update all the dependencies we can

## Discussion
- Largest update is to Svelte Kit v2
- Leaving eslint on v8 still
- rc release of API-Augment so we can update all the Polkadotjs packages.

## Checklist

- [x] PR Self-Review and Commenting
- [x] Docs updated
- [x] Tests added

## How To Test the Changes

1. Clone the pr branch
2. Run the normal `npm test`, `npm run dev`, `npm run build`

### CODE REVIEW

File: src/components/DropDownMenu.svelte [hunk 1]
```diff
@@ -1,8 +1,8 @@
 <script lang="ts" generics="T">
   export let label: string;
   export let id: string = '';
-  export let options: T[];
-  export let value: T | null = null;
+  export let options: T[]; // eslint-disable-line no-undef
+  export let value: T | null = null; // eslint-disable-line no-undef
```
Comment (wilwade) [inline src/components/DropDownMenu.svelte:5]: eslint doesn't know how to handle svelte generics

File: svelte.config.js [hunk 2]
```diff
@@ -1,4 +1,4 @@
-import { vitePreprocess } from '@sveltejs/kit/vite';
+import { vitePreprocess } from '@sveltejs/vite-plugin-svelte';
```
Comment (wilwade) [inline svelte.config.js:1]: Per migration docs

File: svelte.config.js [hunk 3]
```diff
@@ -20,8 +20,16 @@ const config = {
     }),
     paths: {
       base: dev ? '' : process.env.BASE_PATH,
-    }
-  }
+    },
+    alias: {
```
Comment (wilwade) [inline svelte.config.js:24]: These are where these live now

File: src/components/StakeForm.svelte [hunk 4]
```diff
@@ -4,21 +4,16 @@
...
-  import { onMount } from 'svelte';
   import { formatAccount, getExtension } from '$lib/utils';
   import DropDownMenu from './DropDownMenu.svelte';
   import { type Account, allAccountsStore } from '$lib/stores/accountsStore';
 
   export let close: () => void;
   export let stakeAmount: bigint = 1n;
 
-  let selectedAccount: Account;
+  let selectedAccount: Account | null = $allAccountsStore.get($user.address) || null;
```
Comment (wilwade) [inline src/components/StakeForm.svelte:14]: This doesn't need to be onMount, and this fixes the flaky test

File: src/lib/stores/storable.ts
Comment (wilwade) [inline src/lib/stores/storable.ts:1]: Small cleanup

File: src/lib/utils.ts [hunk 5]
```diff
@@ -110,6 +110,7 @@ export const getExtension = async (account: Account) => {
         return await thisWeb3FromSource(account.injectedAccount.meta.source);
       }
     }
+    console.error('No wallet extensions found!');
```
Comment (wilwade) [inline src/lib/utils.ts:113]: Log error out if we don't have any.

File: vitest.config.ts [hunk 6]
```diff
@@ -19,5 +19,8 @@ export default defineConfig({
     environment: 'jsdom',
     globals: true,
     reporters: 'basic',
+    coverage: {
+      include: ['src/**'],
+    },
```
Comment (wilwade) [inline vitest.config.ts:24]: Don't need all those extra files in the coverage report

**Review by claireolmstead — APPROVED** [review]
LGTM. Nice updates using `?.` instead of `val && val...` and other null value error handling.

**Pull request merged**